nobl9/metadata_schema.go

```diff
@@ -51,15 +82,61 @@ func schemaDescription() *schema.Schema {
 	}
 }
 
-func marshalMetadata(d *schema.ResourceData) n9api.MetadataHolder {
+func marshalMetadata(d *schema.ResourceData) (n9api.MetadataHolder, diag.Diagnostics) {
+	var diags diag.Diagnostics
+
+	var labels []interface{}
+	if labelsData := d.Get("labels"); labelsData != nil {
+		labels = labelsData.([]interface{})
+	}
+	var labelsMarshalled n9api.Labels
+	labelsMarshalled, diags = marshalLabels(labels)
+
 	return n9api.MetadataHolder{
 		Metadata: n9api.Metadata{
 			Name:        d.Get("name").(string),
 			DisplayName: d.Get("display_name").(string),
 			Project:     d.Get("project").(string),
-			// TODO Metadata should also support labels - SDK is outdated
+			Labels:      labelsMarshalled,
 		},
+	}, diags
+}
+
+type label struct {
+	key   string
+	value string
+}
+
+func newLabel(labelRaw string) (*label, error) {
+	labelSegments := strings.Split(labelRaw, ":")
+	if len(labelSegments) != 2 {
+		return nil, fmt.Errorf("wrong label format, expected \"key:value\", got \"%s\" ", labelRaw)
 	}
+	return &label{
+		key:   labelSegments[0],
+		value: labelSegments[1],
+	}, nil
+}
+
+func (l label) toString() string {
```

Comment:
I think the good golang practice is to implement stringer interface in that case, wdyt?

```suggestion
func (l label) String() string {
```

Reply:
true :)